apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ClientPool.cs

```diff
@@ -45,9 +44,10 @@ internal ClientPool(ISpannerClientFactory clientFactory = null)
 
         public async Task<SpannerClient> AcquireClientAsync(
             ITokenAccess credentials = null,
-            ServiceEndpoint endpoint = null)
+            ServiceEndpoint endpoint = null,
+            IDictionary additionalOptions = null)
```

Comment:
I'm going to with it as is and when I revisit to move all options into this pattern, I will alter the key to simply be the connectionstringbuilder itself (which is always required and built).
Its a little wasteful, but technically correct.